list.go

```diff
@@ -190,11 +303,15 @@ func (d *Decoder) readTypedList(tag byte) (interface{}, error) {
 			if it != nil {
 				aryValue = reflect.Append(aryValue, EnsureRawValue(it))
 			} else {
-				aryValue = reflect.Append(aryValue, NilValue)
+				aryValue = reflect.Append(aryValue, reflect.Zero(aryValue.Type().Elem()))
```

Comment:
Ask a question: does this code can solve the problem [how to append nil to dynamic type slice by reflect.Append](https://stackoverflow.com/questions/56396510/how-to-append-nil-to-dynamic-type-slice-by-reflect-append) like the code below:

https://github.com/dubbogo/hessian2/blob/f89e95fc44266f9955143c86e95756e9ea6ba9cb/list.go#L358-L362

Reply:
`NilValue`  applicable to untyped. But it is` *interface `. When typed, We need accurate type.

Reply:
@fangyincheng So can we also use `reflect.Zero(aryValue.Type().Elem())` to replace `NilValue` when in untyped situation?

Reply:
I will try it.

Reply:
@wongoo @fangyincheng I think `reflect.Zero(aryValue.Type().Elem())` is a nice work. It is better than `NilValue` because `aryValue.Type().Elem()` is more precise and unified than the `reflect.TypeOf((*interface{})(nil)).Elem()` to represent a `nil` element in `[]interface{}`.

Reply:
@gaoxinge agree!

Reply:
@gaoxinge  OK.